src/Library/NewsWidget.php

```diff
@@ -156,7 +156,12 @@ public function getPosts( $limit ) {
 		$body  = wp_remote_retrieve_body( $request );
 		$posts = json_decode( $body );
 
-		if ( $limit ) {
+		/*
+		 * If we're limiting the number of posts, do so now.
+		 *
+		 * Validate posts with is_array() because array_slice() expects an array as the first param.
+		 */
+		if ( $limit && is_array( $posts ) ) {
 			$posts = array_slice( $posts, 0, $limit );
 		}
```

Comment:
If `$posts` is not an array, then it looks like the return will be false.  The `@return` says `array`.  The method appears to be used only in this file, but the method is public so it may be used by others who expect the return type to be `array`.  You should cast the return as an array or use an `else` here to set `$posts` as an empty array.

Reply:
Changes made @cssjoe - please see https://github.com/BoldGrid/library/commit/37a2be168aba6372b0e7005fca6742bda722e2bc

Reply:
A simple typecast (`(array)`) will work.

Reply:
How about these apples https://github.com/BoldGrid/library/commit/faa4f92e8e0bcb9f235f41fe5b3af1dfde1037f8

Reply:
The entire fix is simply https://github.com/BoldGrid/library/pull/84/files